vault/src/collateral.rs

```diff
@@ -48,19 +48,25 @@ pub async fn maintain_collateralization_rate(
 pub async fn lock_required_collateral<P: VaultRegistryPallet + CollateralBalancesPallet>(
     provider: P,
     vault_id: AccountId,
-    maximum_collateral: u128,
+    maximum_collateral: Option<u128>,
 ) -> Result<(), Error> {
     // check that the vault is registered and active
     let vault = provider.get_vault(vault_id.clone()).await?;
     if !matches!(vault.status, VaultStatus::Active(..)) {
         return Err(Error::RuntimeError(runtime::Error::VaultNotFound));
     }
 
-    let (required_collateral, actual_collateral) = future::try_join(
-        provider.get_required_collateral_for_vault(vault_id),
-        provider.get_reserved_balance(),
-    )
-    .await?;
+    let actual_collateral = vault.backing_collateral;
+
+    let (required_collateral, maximum_collateral) =
+        future::try_join(provider.get_required_collateral_for_vault(vault_id), async {
+            if let Some(max) = maximum_collateral {
+                Ok(max)
+            } else {
+                provider.get_free_balance().await
```

Comment:
This would leave the vault without funds for paying transaction costs

Reply:
Note that we wouldn't necessarily use the whole available free balance if it isn't required to maintain secure collateralization.

Reply:
I don't see a particularly good way to account for this, we have the same problem is `max_collateral` is too high.

Reply:
It's problematic.. I'm ok with merging now. Can you make a ticket for the issue though? It's definitely something we should look into